packages/gateway/src/gateway-language-model.ts

```diff
@@ -40,79 +45,112 @@ export class GatewayLanguageModel implements LanguageModelV2 {
+        warnings: [],
+      };
+    } catch (error) {
+      throw this.handleError(error);
+    }
   }
 
   async doStream(
     options: Parameters<LanguageModelV2['doStream']>[0],
   ): Promise<Awaited<ReturnType<LanguageModelV2['doStream']>>> {
     const { abortSignal, ...body } = options;
 
-    const { value: response, responseHeaders } = await postJsonToApi({
-      url: this.getUrl(),
-      headers: combineHeaders(
-        await resolve(this.config.headers()),
-        options.headers,
-        this.getModelConfigHeaders(this.modelId, true),
-        this.config.o11yHeaders,
-      ),
-      body: this.maybeEncodeFileParts(body),
-      successfulResponseHandler: createEventSourceResponseHandler(z.any()),
-      failedResponseHandler: createJsonErrorResponseHandler({
-        errorSchema: z.any(),
-        errorToMessage: data => data,
-      }),
-      ...(abortSignal && { abortSignal }),
-      fetch: this.config.fetch,
-    });
-
-    return {
-      stream: response.pipeThrough(
-        new TransformStream<
-          ParseResult<LanguageModelV2StreamPart>,
-          LanguageModelV2StreamPart
-        >({
-          transform(chunk, controller) {
-            if (chunk.success) {
-              controller.enqueue(chunk.value);
-            } else {
-              controller.error(
-                (chunk as { success: false; error: unknown }).error,
-              );
-            }
-          },
+    try {
+      const { value: response, responseHeaders } = await postJsonToApi({
+        url: this.getUrl(),
+        headers: combineHeaders(
+          await resolve(this.config.headers()),
+          options.headers,
+          this.getModelConfigHeaders(this.modelId, true),
+          this.config.o11yHeaders,
+        ),
+        body: this.maybeEncodeFileParts(body),
+        successfulResponseHandler: createEventSourceResponseHandler(z.any()),
+        failedResponseHandler: createJsonErrorResponseHandler({
+          errorSchema: z.any(),
+          errorToMessage: data => data,
         }),
-      ),
-      request: { body },
-      response: { headers: responseHeaders },
-    };
+        ...(abortSignal && { abortSignal }),
+        fetch: this.config.fetch,
+      });
+
+      return {
+        stream: response.pipeThrough(
+          new TransformStream<
+            ParseResult<LanguageModelV2StreamPart>,
+            LanguageModelV2StreamPart
+          >({
+            transform(chunk, controller) {
+              if (chunk.success) {
+                controller.enqueue(chunk.value);
+              } else {
+                controller.error(
+                  (chunk as { success: false; error: unknown }).error,
+                );
+              }
+            },
+          }),
+        ),
+        request: { body },
+        response: { headers: responseHeaders },
+      };
+    } catch (error) {
+      throw this.handleError(error);
+    }
+  }
+
+  private handleError(error: unknown) {
```

Comment:
- prefer function (not a method)
- name suggestion: `asGatewayError` or `createGatewayError`